app/scripts/app.js

```diff
@@ -485,6 +485,118 @@ angular
+        if (isNaN(lastInteraction)) {
+          Logger.warn("Last interaction timestamp has been corrupted. The logout timeout will be restarted.");
+          lastInteraction = new Date();
+        }
+        var currentTime = new Date();
+        // Since the INACTIVITY_TIMEOUT_PERIOD is set in minutes it needs to be converted into milliseconds.
+        if (currentTime - lastInteraction > Constants.INACTIVITY_TIMEOUT_PERIOD * 60000) {
+          showLogoutModal();
+        }
+      }, 60000);
+    };
+
+    // In case of multiple opened tabs the interval needs to be restarted everytime there is any activity in any tab,
+    // so all intervals are synced, and they start the counting at the same time.
+    var restartCheckInteractionInterval = function() {
+      // Handles cases of user activity during the opened modal, where user clicks inside the modal but not on the buttons.
+      if (modalInstance) {
+        modalInstance.dismiss("User activity");
+        modalInstance = null;
+      }
+      clearInterval(checkInteractionInterval);
+      startInteractionIntervalCheck();
+    };
+    // Updates lastInteraction date and restarts interval
+    var updateLastInteraction = function() {
+      restartCheckInteractionInterval();
+      localStorage.setItem(lastInteractionKey, new Date().toString());
+    };
+    // Updates Local Storage variable that indicates if all opened tabs should log out current user.
+    var logoutAllTabs = function(bool) {
+      localStorage.setItem(inactivityLogoutKey, bool.toString());
+    };
+    // Logout with passing reason.
+    var logout = function() {
+      var logoutURI = URI.expand("/logout{?cause*}", {
+        cause: 'timeout'
+      });
+      $location.url(logoutURI.toString());
+    };
+    // Show inactivity logout modal.
+    // Before the the modal is shown the handler for checking the changes in local storage is detached.
+    // After the the modal is closed or dismissed the handler will be attached again. This is because of 
+    // the fact that the handler is attached 
+    function showLogoutModal() {
+      if (isLocalLogoutModalShown) {
+        return false;
+      }
+      isLocalLogoutModalShown = true;
+      modalInstance = $uibModal.open({
+        animation: true,
+        templateUrl: 'views/modals/logout.html',
+        controller: 'LogoutModalController',
+      });
+      modalInstance.result.then(function(event) {
+        if (event === 'logout') {
+          logoutAllTabs(true);
+          logout();
+        } else if (event === 'cancel') {
+          // In case of multiple tabs, when the logoutModal would be canceled, the timer has to be restarted.
+          updateLastInteraction();
+          isLocalLogoutModalShown = false;
+        }
+      }, function () {
+        updateLastInteraction();
+        isLocalLogoutModalShown = false;
+      });
+    }
+
+    // Set interval for all timeRemainingFromNow directives.
+    setInterval(function() {
```

Comment:
I don't want this in this block, since this block is not run unless the inactivity timeout feature is enabled.  The directive / filter are independent of that feature. Just take the content of this function and add it to the existing interval that is handling the duration timestamps

Reply:
So I would expect it to be part of this same interval https://github.com/openshift/origin-web-console/blob/master/app/scripts/app.js#L504